LanguageFeatures/Patterns/shared_case_scope_A02_t02.dart

```diff
@@ -65,9 +67,10 @@ String test2(int x) {
 
 String test3(num? x) {
   switch (x) {
-    case var a! when a == 0:
-    case num a when a == 1:
-    case num a? when a == 2:
+    case var a? when a == 0:
```

Comment:
I think flow analysis is going to keep `a` on this branch at `num?`. The reason why it won't promote `a` to `num` is that an equality test is a rather weak witness of anything (because it could in principle be an arbitrary piece of code that says 'true' when you ask whether a particular object is equal to 0, or indeed anything).

You could say that this is special because `num?` admits only null, `int`s, and `double`s, and they all have equality which is computable at compile time (except that we don't know the value of `a`, of course). So we could in principle rule out null, because `null == 0` will definitely yield false.

However, I just don't think the flow analysis will do any of these things. Here is a concrete example yielding a strong hint that the flow analysis doesn't do it:

```dart
void main() {
  int? a = 1 as dynamic;
  if (a == 0) {
    a.isEven; // Error: `a` could be null.
  }
}
```